src/gui/src/heatMap.cpp

```diff
@@ -88,6 +84,41 @@ void HeatMapDataSource::registerHeatMap()
   Gui::get()->registerHeatMap(this);
 }
 
+void HeatMapDataSource::dumpToFile(const std::string& file)
```

Comment:
Can this be lifted out of the gui module? It seems like it would be more generally useful for command line only modes.

Reply:
I seems like it would be a good idea and would make sense to do so at some point in the future.

Reply:
Some of the GUI commands are available in command line mode like save_image (we use the offscreen renderer in that case).  This could be another or it could be moved out.

Reply:
@maliberty at it's core, the heatmap could work without the GUI, and the core parts (the ones that don't rely on the GUI, could be moved) I would say that would be a longer term project. But we could make a JIRA for it if @QuantamHD thinks it would be useful.